use-cases/NIC-passthrough-and-Kata.md

```diff
@@ -0,0 +1,292 @@
+
+\-    [How to install Docker]( https://github.com/kata-containers/documentation/blob/master/install/docker/ubuntu-docker-install.md) 
+
+
+
+## Pass a physical NIC into a Kata Container 
+
+### Part 1 – The Host
+
+**Note:** all the next commands are run as a super-user. Use `$ sudo -i` to become a SU.
+
+
+
+#### 1. Configure system to load with IOMMU support.
+
+   Edit grub file ( */etc/default/grub* )
+
+```bash
+vi /etc/default/grub
+```
+
+And add ` intel_iommu=on iommu=pt` To the **GRUB_CMDLINE_LINUX** line 
+
+> GRUB_CMDLINE_LINUX=" … quiet intel_iommu=on iommu=pt"
+
+
+
+If using UEFI: (like I do) 
+
+/TODO – Change to Ubuntu
+
+```bash
+grub2-mkconfig -o /boot/efi/EFI/centos/grub.cfg
+```
+
+Else (not using UEFI):
+
+```bash
+grub2-mkconfig -o /boot/grub2/grub.cfg
+```
+
+ > Expected result
+ >
+ > Generating grub configuration file ...
+ >
+ > Found linux image: /boot/vmlinuz-3.10.0-957.el7.x86_64
+ >
+ > Found initrd image: /boot/initramfs-3.10.0-957.el7.x86_64.img
+ >
+ > Found linux image: /boot/vmlinuz-0-rescue-dcd5b66ae0534f86a683928888d3f7aa
+ >
+ > Found initrd image: /boot/initramfs-0-rescue-dcd5b66ae0534f86a683928888d3f7aa.img
+ >
+ > done
+
+
+
+#### 2. Test that the system loads with IOMMU:
+
+```bash
+dmesg | head 
+```
+/TODO – Change to Ubuntu 
+
+> Expected results 
+>
+> [    0.000000] Command line: BOOT_IMAGE=/vmlinuz-3.10.0-957.el7.x86_64 root=/dev/mapper/centos-root ro crashkernel=auto [rd.lvm.lv](http://rd.lvm.lv)=centos/root [rd.lvm.lv](http://rd.lvm.lv)=centos/swap rhgb quiet LANG=en_US.UTF-8 intel_iommu=on iommu=pt
+
+
+```bash
+dmesg | grep -ie iommu
+```
+
+> Expected results
+>
+> ...
+>
+> [    1.081716] iommu: Adding device 0000:08:00.0 to group 33 
+>
+> [    1.081736] iommu: Adding device 0000:09:00.0 to group 34 
+>
+> [    1.081755] iommu: Adding device 0000:0a:00.0 to group 35 
+>
+> [    1.081776] iommu: Adding device 0000:0b:00.0 to group 36 
+>
+> [    1.081795] iommu: Adding device 0000:0c:00.0 to group 37 
+>
+> ...
+
+
+
+#### 3. Enable the *hotplug_vfio_on_root_bus* configuration in the *kata configuration.toml* file:
+
+```bash
+vi /usr/share/defaults/kata-containers/configuration.toml
+```
+a. Set hotplug_vfio_on_root_bus = true
+
+b. Make sure you are using the PC machine type by verifying machine_type = "pc"
+
+
+
+#### 4. Install *lshw* and *pciutils* (Ubuntu)
+
+```bash
+apt install lshw pciutils
+```
+
+
+
+#### 5. Find PCI address for the interface you wish to pass to the container
+
+Say I want to pass **enp0s20f1** to the container (see what interfaces you have by running `# ip link` ). 
+
+Get address: 
+
+```bash
+lshw -class network -businfo -numeric | grep enp0s20f1
+```
+
+> Expected results 
+> (something like)
+> 
+> pci@0000:00:14.1 enp0s20f1 network Ethernet Connection I354 1.0 GbE Backplane
+
+
+
+#### 6. Insert the address into a variable
+
+```bash
+BDF="0000:00:14.1"
```

Comment:
This could be  assigned from the `lshw` output with the help of grep and awk and other parsing help something like :
`BDF=$(shw -class network -businfo -numeric | grep ${NIC} | awk 'print $1' | cut -d@ -f2)`